ui/packages/catalog-ui-search/src/main/webapp/react-component/container/results-export/results-export.tsx

```diff
@@ -0,0 +1,184 @@
+import ResultsExportComponent from '../../presentation/results-export'
+import { exportResult, exportResultSet } from '../../utils/export'
+import { getResultSetCql } from '../../utils/cql'
+import saveFile from '../../utils/save-file'
+import withListenTo, { WithBackboneProps } from '../backbone-container'
+
+const contentDisposition = require('content-disposition')
+
+type Result = {
+  id: number
+  source: string
+}
+
+type ExportFormat = {
+  id: string
+  displayName: string
+}
+
+type Props = {
+  store: any
+} & WithBackboneProps
+
+type State = {
+  downloadDisabled: boolean
+  selectedFormat: string
+  exportFormats: ExportFormat[]
+  selectedResults: Result[]
+}
+
+class ResultsExport extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      selectedFormat: 'Select an export option',
+      exportFormats: [],
+      downloadDisabled: true,
+      ...this.mapSelectionToState(),
+    }
+    this.props.listenTo(
+      this.props.store.getSelectedResults(),
+      'update add remove reset',
+      this.handleSelectionChange
+    )
+  }
+  handleSelectionChange = () => {
+    this.setState(this.mapSelectionToState())
+  }
+  mapSelectionToState = () => {
+    const selectedResults = this.props.store
+      .getSelectedResults()
+      .toJSON()
+      .map((result: any) => {
+        return {
+          id: result['metacard']['id'],
+          source: result['metacard']['properties']['source-id'],
+        }
+      })
+    return {
+      selectedResults,
+    }
+  }
+  componentDidUpdate(_prevProps: Props, prevState: State) {
+    if (prevState.selectedResults !== this.state.selectedResults) {
+      this.fetchExportOptions()
+    }
+  }
+  componentDidMount() {
+    this.fetchExportOptions()
+  }
+  fetchExportOptions = () => {
+    let transformerType = 'metacard'
+
+    if (this.state.selectedResults.length > 1) {
+      transformerType = 'query'
+    }
+
+    fetch(`./internal/transformers/${transformerType}`)
+      .then(response => response.json())
+      .then((exportFormats: ExportFormat[]) => {
+        return exportFormats.sort(
+          (format1: ExportFormat, format2: ExportFormat) => {
+            if (format1.displayName > format2.displayName) {
+              return 1
+            }
+
+            if (format1.displayName < format2.displayName) {
+              return -1
+            }
+
+            return 0
+          }
+        )
+      })
+      .then(exportFormats =>
+        this.setState({
+          exportFormats: exportFormats,
+        })
+      )
+  }
+  getResultSources() {
+    return new Set(this.state.selectedResults.map(result => result.source))
+  }
+  getSelectedExportFormatId() {
+    const selectedFormat = this.state.selectedFormat
+    const format = this.state.exportFormats.find(
+      format => format.displayName === selectedFormat
+    )
+
+    if (format !== undefined) {
+      return format.id
+    }
+
+    return undefined
```

Comment:
✏️ Wouldn't it be the same to just always return `format.id`?

Reply:
<img width="608" alt="screen shot 2019-02-20 at 2 21 06 pm" src="https://user-images.githubusercontent.com/7646577/53125267-d629db80-351a-11e9-890c-0511b5068880.png">

Reply:
It took me a minute to understand why this doesn't work, but I got it now. Can't call `.id` on undefined so can't really return it. That's annoying!